pkg/webhook/mutation/pod/oneagent/mutator.go

```diff
@@ -112,6 +112,37 @@ func (mut *Mutator) ensureInitSecret(request *dtwebhook.MutationRequest) error {
 	return nil
 }
 
+func (mut *Mutator) isInjectionPossible(request *dtwebhook.MutationRequest) (bool, string) {
+	reasons := []string{}
+
+	status := request.DynaKube.Status
+
+	_, err := request.DynaKube.TenantUUIDFromConnectionInfoStatus()
+	if err != nil {
+		log.Info("tenant UUID is not available, OneAgent cannot be injected", "pod", request.PodName())
+
+		reasons = append(reasons, EmptyTenantUUIDReason)
+	}
+
+	if !request.DynaKube.IsOneAgentCommunicationRouteClear() {
+		log.Info("OneAgent communication route is not clear, OneAgent cannot be injected", "pod", request.PodName())
+
+		reasons = append(reasons, EmptyConnectionInfoReason)
+	}
+
+	if status.CodeModules.Version == "" {
+		log.Info("code modules version is not available, OneAgent cannot be injected", "pod", request.PodName())
```

Comment:
```suggestion
		log.Info("codemodules version is not available, OneAgent cannot be injected", "pod", request.PodName())
```

Reply:
minor nitpick because of consistency

Apart from that my only complaint would align with @0sewa0 comment about the codemodules case.

Reply:
Updated